src/vmm/src/vstate.rs

```diff
@@ -1245,13 +1267,17 @@ mod tests {
         // Stop it by sending exit.
         assert!(vcpu_handle.send_event(VcpuEvent::Exit).is_ok());
 
-        // Validate vCPU thread ends execution.
-        vcpu_handle
-            .join_vcpu_thread()
-            .expect("failed to join thread");
-
-        // Validate that the vCPU signaled its exit.
-        assert_eq!(vcpu_exit_evt.read().unwrap(), 1);
+        // Sending exit won't stop the thread.
+        // Instead of waiting the completion of the thread, we need to poll the fd.
+        for _i in 0..10 {
+            match vcpu_exit_evt.read() {
+                Ok(x) => assert_eq!(x, 1),
+                Err(e) if e.kind() == io::ErrorKind::WouldBlock => {
+                    thread::sleep(Duration::from_millis(100))
+                }
+                Err(e) => panic!("failed to read the exit event: {}", e),
+            }
+        }
```

Comment:
Using `queue_event_expect_response(&vcpu_handle, VcpuEvent::Exit, VcpuResponse::Paused)` is another way. It *looks* nicer than the for loop.

However `queue_event_expect_response()` is calling `recv_timeout(Duration::from_millis(100))`. So it doesn't address the concern actually.